### PR TITLE
Remove deprecated decorators from test_base

### PR DESCRIPTION
**Pull request by adriangb**

### CODE REVIEW

File: tests/middleware/test_base.py [hunk 1]
```diff
@@ -82,20 +82,21 @@ def test_custom_middleware(test_client_factory):
 
 
 def test_middleware_decorator(test_client_factory):
```
Comment (aminalaee) [inline tests/middleware/test_base.py]: Do we need to keep this test?
As the name suggests I think we're not really testing what it says. I tried to remove that in #1499 but I'll close it in favour of this.

Comment (lovelydinosaur) [reply tests/middleware/test_base.py]: > Do we need to keep this test?

I don't think so, no.

Comment (adriangb) [reply tests/middleware/test_base.py]: removed 😄

File: tests/middleware/test_base.py [hunk 2]
```diff
@@ -80,31 +80,6 @@ def test_custom_middleware(test_client_factory):
...
-    client = test_client_factory(app)
-    response = client.get("/")
-    assert response.text == "OK"
-
-    response = client.get("/homepage")
-    assert response.text == "Homepage"
-    assert response.headers["Custom"] == "Example"
-
-
 def test_state_data_across_multiple_middlewares(test_client_factory):
```
Comment (aminalaee) [inline tests/middleware/test_base.py:84]:
```suggestion

def test_state_data_across_multiple_middlewares(test_client_factory):
```

**Comment by aminalaee** [comment]
I was expecting when this would happen 😄 The methods deprecated removed need the `no cover`.

**Comment by adriangb** [comment]
Yup added the no covers

**Review by aminalaee — APPROVED** [review]

**Pull request merged**